mir-ci/mir_ci/robot_libraries/WaylandHid.py

```diff
@@ -19,44 +20,90 @@ class WaylandHid(VirtualPointer):
 
     ROBOT_LISTENER_API_VERSION = 3
 
-    def __init__(self) -> None:
+    def __init__(self, pixel_step=16) -> None:
+        """
+        Initialize the WaylandHid instance.
+
+        :param pixel_step: Step size in pixels to be used when a nonpositive
+                           number of steps is passed to walk_pointer_*.
+                           Default is 16.
+        """
         self.ROBOT_LIBRARY_LISTENER = self
+        self._pointer_position: Optional[Tuple[int, int]] = None
+        self._pixel_step = pixel_step
         display_name = os.environ.get("WAYLAND_DISPLAY", "wayland-0")
         super().__init__(display_name)
 
     @keyword
-    async def move_pointer_to_absolute(self, x: float, y: float) -> None:
-        """Move the virtual pointer to an absolute position within the output."""
+    async def move_pointer_to_absolute(self, x: int, y: int) -> None:
         await self.connect()
         self.move_to_absolute(x, y)
+        self._pointer_position = (x, y)
 
     @keyword
-    async def move_pointer_to_proportional(self, x: float, y: float) -> None:
-        """Move the virtual pointer to a position proportional to the size of the output."""
+    async def move_pointer_to_proportional(self, x: float, y: float) -> tuple[int, int]:
         await self.connect()
         self.move_to_proportional(x, y)
+        absolute_position = self.get_absolute_from_proportional(x, y)
+        self._pointer_position = absolute_position
+        return absolute_position
+
+    @keyword
+    async def walk_pointer_to_absolute(self, x: int, y: int, steps: int, delay: float) -> None:
+        await self.connect()
+        assert self._pointer_position is not None, "Cannot walk without moving the pointer at least once"
+        assert self._pixel_step > 0, "pixel_step must be greater than 0"
+        if steps <= 0:
+            distance = ((self._pointer_position[0] - x) ** 2 + (self._pointer_position[1] - y) ** 2) ** 0.5
+            steps = math.ceil(distance / self._pixel_step)
 
-    @keyword("Press ${button} Button")
+        for step in range(0, steps + 1):
+            t = step / steps
+            walk_x = (1 - t) * self._pointer_position[0] + t * x
+            walk_y = (1 - t) * self._pointer_position[1] + t * y
+            self.move_to_absolute(walk_x, walk_y)
+            await asyncio.sleep(delay)
+
+        self._pointer_position = (x, y)
```

Comment:
This is handled by `move_to_absolute` already.
```suggestion
```

Reply:
Not really. `move_to_absolute` is from `VirtualPointer`, not the robot library.